source/tests/consistent/common.py

```diff
@@ -0,0 +1,269 @@
+                graph.as_graph_def(),
+                [f"o_{ii}_{self.unique_id1}" for ii, _ in enumerate(ret1)],
+            )
+            with tf.Graph().as_default() as new_graph:
+                tf.import_graph_def(output_graph_def, name="")
+            tf_obj1.init_variables(new_graph, output_graph_def, suffix=self.unique_id1)
+
+            data = tf_obj1.serialize(suffix=self.unique_id1)
+        dp_obj = self.dp_class.deserialize(data)
+        ret2 = self.eval_dp(dp_obj)
+        self.compare_tf_dp_ret(ret1, ret2)
+
+    def test_tf_self_consistent(self):
+        """Test whether TF is self consistent."""
+        tf_obj1 = self.init_tf()
+        with tf.Session(config=default_tf_session_config) as sess:
+            graph = tf.get_default_graph()
+            ret1 = self.build_eval_tf(sess, tf_obj1, suffix=self.unique_id1)
+            output_graph_def = tf.graph_util.convert_variables_to_constants(
+                sess,
+                graph.as_graph_def(),
+                [f"o_{ii}_{self.unique_id1}" for ii, _ in enumerate(ret1)],
+            )
+            with tf.Graph().as_default() as new_graph:
+                tf.import_graph_def(output_graph_def, name="")
+            tf_obj1.init_variables(new_graph, output_graph_def, suffix=self.unique_id1)
+
+            data = tf_obj1.serialize(suffix=self.unique_id1)
+            tf_obj2 = self.tf_class.deserialize(data, suffix=self.unique_id2)
+            ret2 = self.build_eval_tf(sess, tf_obj2, suffix=self.unique_id2)
+        for rr1, rr2 in zip(ret1, ret2):
+            np.testing.assert_allclose(rr1, rr2)
+
+    def test_dp_self_consistent(self):
+        """Test whether DP is self consistent."""
+        if self.skip_dp:
+            self.skipTest("Unsupported")
+        tf_obj1 = self.init_tf()
+        with tf.Session(config=default_tf_session_config) as sess:
+            graph = tf.get_default_graph()
+            ret1 = self.build_eval_tf(sess, tf_obj1, suffix=self.unique_id1)
+            output_graph_def = tf.graph_util.convert_variables_to_constants(
+                sess,
+                graph.as_graph_def(),
+                [f"o_{ii}_{self.unique_id1}" for ii, _ in enumerate(ret1)],
+            )
+            with tf.Graph().as_default() as new_graph:
+                tf.import_graph_def(output_graph_def, name="")
+            tf_obj1.init_variables(new_graph, output_graph_def, suffix=self.unique_id1)
+
+            data = tf_obj1.serialize(suffix=self.unique_id1)
+        dp_obj = self.dp_class.deserialize(data)
+        ret2 = self.eval_dp(dp_obj)
+        data2 = dp_obj.serialize()
+        dp_obj2 = self.dp_class.deserialize(data2)
+        ret3 = self.eval_dp(dp_obj2)
+        for rr2, rr3 in zip(ret2, ret3):
+            if isinstance(rr2, np.ndarray) and isinstance(rr3, np.ndarray):
+                np.testing.assert_allclose(rr2, rr3)
+            else:
+                self.assertEqual(rr2, rr3)
+
+    def test_tf_pt_consistent(self):
```

Comment:
should'nt the cross backend test be designed as test_dp_xxx_consistent?

Reply:
After refactoring, the reference backend is dynamic, according to whether they are available.

https://github.com/deepmodeling/deepmd-kit/blob/a3b7e0f2d778892f101fe3f8888cfc89ca255eb5/source/tests/consistent/common.py#L191-L198